front/app/components/admin/PostManager/components/InitiativesCount.tsx

```diff
@@ -19,84 +16,44 @@ const Container = styled.div`
   font-weight: 500;
 `;
 
-interface InputProps {
-  assignee?: string | null;
-  topics?: string[] | null;
-  initiativeStatus?: string | null;
-  feedbackNeeded: boolean;
-  searchTerm?: string | null;
-}
-
-interface DataProps {
-  initiativesCount: GetInitiativesCountChildProps;
+interface Props {
+  // We are using initiatives, not initiatives_count query parameter types.
+  // This is because the InitiativesCount component is used in the PostManager,
+  // which uses the initiatives query parameter types.
+  queryParameters: IQueryParameters;
 }
 
-interface Props extends InputProps, DataProps {}
-
-interface State {}
+const InitiativesCount = ({ queryParameters }: Props) => {
+  const { data: initiativesCount } = useInitiativesCount({
+    feedback_needed: queryParameters.feedback_needed,
+    assignee: queryParameters.assignee,
+    topics: queryParameters.topics,
+    initiative_status: queryParameters.initiative_status,
+    search: queryParameters.search,
+  });
 
-export class InitiativesCount extends React.PureComponent<Props, State> {
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.searchTerm !== this.props.searchTerm) {
-      if (isFunction(this.props.initiativesCount.onChangeSearchTerm)) {
-        this.props.initiativesCount.onChangeSearchTerm(this.props.searchTerm);
-      }
-    }
-  }
+  if (!initiativesCount) return null;
 
-  render() {
-    const { initiativesCount } = this.props;
-    const initiativesMatchingFiltersCount = initiativesCount.count;
+  const initiativesMatchingFiltersCount =
+    initiativesCount.data.attributes.count;
 
-    return (
-      <Container>
-        {/*
+  return (
+    <Container>
+      {/*
           If there are no initiatives, we have an 'empty container' to indicate there are no initiatives matching the filters.
           Hence we only show this count when there's at least 1 initiative.
         */}
-        {!isNilOrError(initiativesMatchingFiltersCount) &&
-          initiativesMatchingFiltersCount > 0 &&
-          (initiativesMatchingFiltersCount === 1 ? (
-            <FormattedMessage {...messages.oneInitiative} />
-          ) : (
-            <FormattedMessage
-              {...messages.multipleInitiatives}
-              values={{ initiativesCount: initiativesMatchingFiltersCount }}
-            />
-          ))}
-      </Container>
-    );
-  }
-}
-
-const Data = adopt({
-  initiativesCount: ({
-    topics,
-    initiativeStatus,
-    assignee,
-    feedbackNeeded,
-    render,
-  }) => {
-    return (
-      <GetInitiativesCount
-        feedbackNeeded={feedbackNeeded === true ? true : undefined}
-        assignee={assignee !== 'all' ? assignee : undefined} // TOCHECK
```

Comment:
As we're using the queryParameters now, this is already dealt with. In AssigneeFilter, we set assignee to undefined when value is 'all'. Note that when assignee is 'unassigned', the count doesn't work for initiatives. It does work in the input manager though. This is already the case on production.